### PR TITLE
IPipelineDumper: Add GetGraphicsShaderBinaryHash

### PR DESCRIPTION
**Pull request by nhaehnle**
Suitable for computing dump hash codes for generating pipeline dumps for graphics pipeline library compiles.

A corresponding driver change is pending.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit f4f6a1a3aee4986d8fd9ff27871d4a8b7af03957
<details><summary>CTS tests (Failed: 0/207163)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 36989/69061 (53.6%)</li>
<li>Failed: 0/69061 (0.0%)</li>
<li>Not Supported: 32072/69061 (46.4%)</li><li>Warnings: 0/69061 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 37209/69039 (53.9%)</li>
<li>Failed: 0/69039 (0.0%)</li>
<li>Not Supported: 31830/69039 (46.1%)</li><li>Warnings: 0/69039 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34228/69063 (49.6%)</li>
<li>Failed: 0/69063 (0.0%)</li>
<li>Not Supported: 34835/69063 (50.4%)</li><li>Warnings: 0/69063 (0.0%)</li>
</ul></details>


</ul></details>

File: tool/dumper/vkgcPipelineDumper.cpp [hunk 1]
```diff
@@ -189,6 +189,22 @@ uint64_t VKAPI_CALL IPipelineDumper::GetShaderHash(const void *moduleData) {
   return MetroHash::compact64(reinterpret_cast<const MetroHash::Hash *>(&shaderModuleData->hash));
 }
 
+// =====================================================================================================================
+// Calculates graphics shader binary hash code.
+//
+// @param pPipelineInfo : Info to build this partial graphics pipeline
```
Comment (amdrexu) [inline tool/dumper/vkgcPipelineDumper.cpp:195]: -> pipelineInfo

File: tool/dumper/vkgcPipelineDumper.cpp [hunk 2]
```diff
@@ -189,6 +189,22 @@ uint64_t VKAPI_CALL IPipelineDumper::GetShaderHash(const void *moduleData) {
...
+// @param stage : The shader stage for which the code is calculated
+uint64_t VKAPI_CALL IPipelineDumper::GetGraphicsShaderBinaryHash(const GraphicsPipelineBuildInfo *pipelineInfo,
```
Comment (amdrexu) [inline tool/dumper/vkgcPipelineDumper.cpp:197]: I am considering if we can completely follow LLPC coding style in such interface files used by driver. The origin coding style transition left this problem because it needs touch driver codes to make those changes. Note sure if we decide to resolve it now. Of course, this issue is unrelated with this change.

**Review by amdrexu — APPROVED** [review]
